### PR TITLE
Upgrade IntelliJ platform template 

### PR DESCRIPTION
**Pull request by philipp-spiess**
This PR is an attempt to update our plugin to the latest version of https://github.com/JetBrains/gradle-intellij-plugin so we can iterate quicker on potential future changes.

Upgrade instructions are documented [here](https://lp.jetbrains.com/gradle-intellij-plugin/) but are mostly focused on Kotlin instead of our existing Groovy config files.

Learnings so far:

- Setter syntax for Groovy also changed 
- We'll need to upgrade the IntelliJ version to at least 2021.3 for M1 compat (https://youtrack.jetbrains.com/issue/CWM-3540#focus=Comments-27-5192513.0-0)

### CODE REVIEW

**Comment by philipp-spiess** [comment]
Managed to get it to build today with the latest intellij platform version which is great 🎉  I could also start it with a recent version of IntelliJ and it seems to work fine!

However there are still "some" issues:

- The test suite is flaky, I had to patch out an interim "wait for port" step that doesn't seem to be working. In the template. this github action is only run on-demand and it wasn't run in a while it seems, so it's possible that we have to make more changes to that file.
- The build script is failing pretty early during `./gradlew listProductsReleases`, specifically:
    ```
    Execution failed for task ':listProductsReleases'.
    > Cannot query the value of task ':listProductsReleases' property 'sinceBuild' because it has no value available.
      The value of this property is derived from:
        - task ':patchPluginXml' property 'sinceBuild'
     ```
    It's unclear as to why to me, I tried multiple ways to set the since-build property correctly but couldn't make it work ([some context](https://plugins.jetbrains.com/docs/intellij/build-number-ranges.html?from=jetbrains.org#intellij-platform-based-products-of-pre-20162-ide-versions)). The new template setup doesn't define these properties in the XML file anymore and instead uses a custom script in the Kotlin config file to make it work. I'll likely try to update our plugin to the same Kotlin config file (if this is somehow possible) next to see if this fixes the issue. The script _does_ work for the template on my machine.

**Comment by philipp-spiess** [comment]
This seems to be in a good shape now and is ready for a review. The only thing that I could not verify is if the automatic release pipeline does work like it should (we'll need to configure some secrets for that first and, well, have something to actually release?) 

The extension itself seems to work as it did before, you can run it with `./gradlew runIde`:

Screenshot 2022-01-26 at 16 36 57

Contrary to the template repo, I configured the tests to run with every pull request as well. For us they only contain a unit test at the moment, if it becomes too slow we can revisit but I don't see a reason for not running them automatically. 

Given the bigger changes regarding the change log there might be some slight formatting differences that we'll have to fix once we release something.

File: src/test/java/CommitViewUriBuilderTest.java [hunk 1]
```diff
@@ -21,7 +21,7 @@ public void testBuild_AllValid() {
         "intellij",
         "1.1");
 
-    String want = "https://www.sourcegraph.com/github.com/sourcegraph/sourcegraph-jetbrains/-/commit/1fa8d5d6286c24924b55c15ed4d1a0b85ccab4d5?editor=JetBrains&version=v1.2.0&utm_product_name=intellij&utm_product_version=1.1";
+    String want = "https://www.sourcegraph.com/github.com/sourcegraph/sourcegraph-jetbrains.git/-/commit/1fa8d5d6286c24924b55c15ed4d1a0b85ccab4d5?editor=JetBrains&version=v1.2.2&utm_product_name=intellij&utm_product_version=1.1";
```
Comment (slimsag) [inline src/test/java/CommitViewUriBuilderTest.java:24]: potential regression or bug here - this wouldn't be a valid Sourcegraph URL with `.git` in it.

**Comment by slimsag** [comment]
Looks pretty great to me, though I'd love to make sure we have someone with more Java experience than me reviewing.

> we'll need to configure some secrets for that first

this should be easy to do, we can get you access to set this up

> have something to actually release

just wanted to point out there's a quite large enterprise customer that has this plugin rolled out to something like ~2k devs.

File: src/test/java/CommitViewUriBuilderTest.java (hunk 1 above)
Comment (coury-clark) [reply src/test/java/CommitViewUriBuilderTest.java:24]: Just checked and the actual feature isn't working with 1.2.2, so likely there is a bug that already existed and just got copied into this test.

Comment (coury-clark) [reply src/test/java/CommitViewUriBuilderTest.java:24]: Updated the issue: https://github.com/sourcegraph/sourcegraph-jetbrains/issues/25#issuecomment-1022512027

File: src/main/resources/META-INF/plugin.xml [hunk 2]
```diff
@@ -1,73 +1,8 @@
 <idea-plugin>
   <id>com.sourcegraph.jetbrains</id>
   <name>Sourcegraph</name>
-  <version>1.2.2</version>
```
[the remark this answers is not in the document]
Comment (philipp-spiess) [reply src/main/resources/META-INF/plugin.xml:4]: Most of the plugin.xml file is now generated dynamically by reading the config files + README + CHANGELOG, see here: https://github.com/sourcegraph/sourcegraph-jetbrains/pull/30/files#diff-c0dfa6bc7a8685217f70a860145fbdf416d449eaff052fa28352c5cec1a98c06R50-R74

I tried to keep using the existing format first but had some issues with some of the release tasks unfortunately. Plus maintaining a CHANGELOG in the project root makes it easier to work with it I think. The version number specifically is set here: https://github.com/sourcegraph/sourcegraph-jetbrains/pull/30/files#diff-3d103fc7c312a3e136f88e81cef592424b8af2464c468116545c4d22d6edcf19R7

**Review by vdavid — APPROVED** [review]
Wow, I like this a lot! Nice work, @philipp-spiess. This should make it a much better experience to work on the plugin later. (That said, my Java is rusty, and I'm not familiar with Gradle, so my review was pretty superficial.)

**Pull request merged**